### PR TITLE
Break example.c into a Synchronous and an Asynchronous Flavors

### PR DESCRIPTION
**Pull request by hazen**
While there is naturally some overlap between the two, developers will probably pick one type or the other.  Making these separate examples will hopefully improve readability.  Also, this (might) simplify changes to upcoming async testing.

### CODE REVIEW

**Comment by bookshelfdave** [comment]
problems during build:

```
./configure && make
...
mv -f examples/.deps/riak_c_example-riak_command.Tpo examples/.deps/riak_c_example-riak_command.Po
gcc -DPACKAGE_NAME=\"riak_c_client\" -DPACKAGE_TARNAME=\"riak_c_client\" -DPACKAGE_VERSION=\"0.5\" -DPACKAGE_STRING=\"riak_c_client\ 0.5\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"riak_c_client\" -DVERSION=\"0.5\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_MEMSET=1 -DHAVE_SOCKET=1 -DHAVE_STRERROR=1 -DHAVE_ARPA_INET_H=1 -DHAVE_STDDEF_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_STRINGS_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDLIB_H=1 -DHAVE_MALLOC=1 -I.  -DUSE_DEBUG -D_RIAK_DEBUG -I./src/include -I./src/internal -I./riak_pb/c -I./src/adapters -I./src -Iexamples/include  -g -Wall -g -O2 -MT examples/riak_c_example-example_log.o -MD -MP -MF examples/.deps/riak_c_example-example_log.Tpo -c -o examples/riak_c_example-example_log.o `test -f 'examples/example_log.c' || echo './'`examples/example_log.c
mv -f examples/.deps/riak_c_example-example_log.Tpo examples/.deps/riak_c_example-example_log.Po
/bin/sh ./libtool  --tag=CC   --mode=link gcc -g -Wall -g -O2   -o riak_c_example examples/riak_c_example-example.o examples/riak_c_example-riak_command.o examples/riak_c_example-example_log.o -lriak_c_client-0.5 -L/usr/local/Cellar/protobuf-c/0.15/lib -lprotobuf-c  -L/usr/local/Cellar/protobuf/2.5.0/lib -lprotobuf -D_THREAD_SAFE
libtool: link: gcc -g -Wall -g -O2 -o .libs/riak_c_example examples/riak_c_example-example.o examples/riak_c_example-riak_command.o examples/riak_c_example-example_log.o -D_THREAD_SAFE  /Users/dparfitt/basho/riak-c-client/.libs/libriak_c_client-0.5.dylib -L/usr/local/Cellar/protobuf-c/0.15/lib -L/usr/local/Cellar/protobuf/2.5.0/lib -L/usr/local/Cellar/libevent/2.0.21/lib -levent_pthreads -levent -lprotobuf-c -lprotobuf
make: *** No rule to make target `examples/async_example.c', needed by `examples/riak_c_async_example-async_example.o'.  Stop.
```

File: examples/example.c [hunk 1]
```diff
@@ -182,46 +108,22 @@
...
         switch (operation) {
         case RIAK_COMMAND_PING:
-            if (args.async) {
-                err = riak_async_register_ping(rop, (riak_response_callback)example_ping_cb);
-            } else {
-                err = riak_ping(cxn);
-                printf("PONG\n");
-            }
+            err = riak_ping(cxn);
+            printf("PONG\n");
```
Comment (bookshelfdave) [inline examples/example.c]: `PONG` should only occur if ! err, otherwise a `PANG`

File: examples/example_log.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * under the License.
+ *
+ *********************************************************************/
+
+#include "riak.h"
+#include "example_log.h"
+
+riak_int32_t
+example_log_init(void *ptr) {
+    example_log_data* datum = (example_log_data*)ptr;
```
Comment (bookshelfdave) [inline examples/example_log.c:28]: can we let the user of the examples know that output will show up in riak.log? Maybe just a simple `printf("riak.log initialized");`

File: examples/example.c (hunk 1 above)
Comment (hazen) [inline examples/example.c]: So do you know the inside joke about `PANG`?

**Comment by bookshelfdave** [comment]
+1

**Pull request merged**